Tools/TransformTool/TransformTool.cs

```diff
@@ -240,6 +242,17 @@ public void ProcessInput(ActionMapInput input, ConsumeControlDelegate consumeCon
 				if (manipulatorGameObject.activeSelf && (hoveringSelection || hasObject))
 					manipulatorGameObject.SetActive(false);
 
+				var hovers = new Dictionary<Transform, GameObject>(m_HoverGameObjects);
+				foreach (var kvp in hovers)
+				{
+					var rayOrigin = kvp.Key;
+					if (!directSelection.ContainsKey(rayOrigin))
```

Comment:
As discussed, let's try for having SelectionTool handle all highlighting and turning TransformTool into an ILinkedTool.